pkg/util/ctxgroup/ctxgroup.go

```diff
@@ -162,9 +169,33 @@ func (g Group) Go(f func() error) {
 	g.wrapped.Go(f)
 }
 
+// ErrTaskPanic is used to wrap a panic thrown by a task in the group. It is
+// generally not returned directly to any call, but may be observed when
+// recovering a panic from Wait(). In such cases, `Payload` represents the
+// exact argument passed to panic() in the task, while wrapping it in an error
+// preserves the original stack in the task before it is captured and re-thrown
+// in Wait().
+type ErrTaskPanic struct {
+	Payload interface{}
+}
+
+func (e ErrTaskPanic) Error() string {
+	return fmt.Sprintf("%v", e.Payload)
+}
+
 // GoCtx calls the given function in a new goroutine.
+// If the function panics before another function has returned an error, calling
+// Wait() will panic with the original panic wrapped in ErrTaskPanic so as to
+// capture its original stack. NB: if another task in the group has already
+// returned an error, the panic will instead be recovered and ignored as the
+// error slot in the underlying errgroup is already used.
```

Comment:
I wonder if, in dev builds we still want to raise this panic somehow?

Reply:
alternatively: https://github.com/cockroachdb/cockroach/commit/1a26d107acf9b9758a6db0fade53783b95d9c431

Reply:
I messed with this a little and now just store the panic myself, along-side the errgroup instead of inlining the errgroup.